marshalling a struct pointer does not work

quick reproducer:

```
package main

import (
  "github.com/naoina/toml"
)

type appcfg struct {
  Debug  bool     `toml:"debug"`
  Admins []string `toml:"admins"`
  API    struct {
    URL       string `toml:"URL"`
    AuthToken string `toml:"authtoken"`
  }
}

func main() {
  str, _ := toml.Marshal(&appcfg{})
  println(str)
}
```

fails with error:

```
panic: reflect: call of reflect.Value.NumField on ptr Value

goroutine 1 [running]:
reflect.flag.mustBe(0x16, 0x19)
        /usr/local/go/src/reflect/value.go:199 +0xa6
reflect.Value.NumField(0x6aa600, 0xc82004ccb0, 0x16, 0x7f99090340a8)
        /usr/local/go/src/reflect/value.go:1146 +0x2a
github.com/naoina/toml.marshal(0x0, 0x0, 0x0, 0x0, 0x0, 0x6aa600, 0xc82004ccb0, 0x16, 0x7f9909030000, 0x0, ...)
        /home/sztanpet/go/gocode/src/github.com/naoina/toml/encode.go:77 +0xca
github.com/naoina/toml.Marshal(0x6aa600, 0xc82004ccb0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/sztanpet/go/gocode/src/github.com/naoina/toml/encode.go:44 +0xc4
github.com/naoina/toml.(*Encoder).Encode(0xc820047da8, 0x6aa600, 0xc82004ccb0, 0x0, 0x0)
        /home/sztanpet/go/gocode/src/github.com/naoina/toml/encode.go:62 +0x47
```

--- a/encode.go
+++ b/encode.go
@@ -74,6 +74,11 @@ type Marshaler interface {
 
 func marshal(buf []byte, prefix string, rv reflect.Value, inArray, arrayTable bool) ([]byte, error) {
 	rt := rv.Type()
+	for rt.Kind() == reflect.Ptr {
+		rv = rv.Elem()
+		rt = rv.Type()
+	}
+
 	for i := 0; i < rv.NumField(); i++ {
 		ft := rt.Field(i)
 		if !ast.IsExported(ft.Name) {
@@ -158,6 +163,13 @@ func encodeValue(buf []byte, prefix, name string, fv reflect.Value, inArray, arr
 			return nil, err
 		}
 		return appendNewline(buf, inArray, arrayTable), nil
+	case reflect.Ptr:
+		newElem := fv.Elem()
+		if newElem.IsValid() {
+			return encodeValue(buf, prefix, name, newElem, inArray, arrayTable)
+		} else {
+			return encodeValue(buf, prefix, name, reflect.New(fv.Type().Elem()), inArray, arrayTable)
+		}
 	}
 	return nil, fmt.Errorf("toml: marshal: unsupported type %v", fv.Kind())
 }

--- a/encode_test.go
+++ b/encode_test.go
@@ -296,3 +296,35 @@ name="plantain"
 		}
 	}
 }
+
+func TestMarshalPointer(t *testing.T) {
+	type Profession struct {
+		Name       string
+		Department string
+	}
+
+	foo := struct {
+		Active     bool
+		Name       string
+		Occupation *Profession
+	}{
+		true,
+		"Foo Bar",
+		&Profession{"Professor", "CS"},
+	}
+
+	expect := `active=true
+name="Foo Bar"
+[occupation]
+name="Professor"
+department="CS"
+`
+	actual, err := toml.Marshal(&foo)
+	if err != nil {
+		t.Errorf(`Unable to marshal pointer, err was %s`, err.Error())
+	}
+
+	if !reflect.DeepEqual(string(actual), expect) {
+		t.Errorf(`Marshal(%#v); v => %#v; want %#v`, foo, string(actual), expect)
+	}
+}

